drivers/usb/udc/udc_dwc2.c

```diff
@@ -18,6 +18,7 @@
 #include <zephyr/sys/byteorder.h>
 #include <zephyr/drivers/usb/udc.h>
 #include <zephyr/usb/usb_ch9.h>
+#include <zephyr/usb/usbd.h>
```

Comment:
You must not include a stack header in the driver, this is mixing different layers. This will never be approved by me. I can show you another way to force a driver to run at full speed if you address my first comment.

Reply:
I reworked your approach and pushed it to https://github.com/jfischer-no/zephyr/commits/pr-76255/. With https://github.com/jfischer-no/zephyr/commit/a9335f66218b94bb19d013a9c096c44052890aa0, the settings in the UDC driver are independent and do not interfere with stack. I also changed stack's Kconfig in https://github.com/jfischer-no/zephyr/commit/df95264e6a50158c922392cbdd38ac8307471226, see the commit message or better try it out.

Reply:
I did try out your changes and it is possible to build High-Speed capable USB device stack with Full-Speed only driver. Such configuration doesn't make sense to me. What's the point? Why make it so complex?

Reply:
It is no more complicated than the original. It always defaults to a valid configuration. The high-speed part always compiles. But yes, it allows you to build a stack that has high-speed support even if the device does not. This is intentional, for example to see the difference in flash/RAM usage.

Reply:
I think that intention is pretty weird. @carlescufi can you please weigh in?